AsusRouterA11yFixes.user.js

```diff
@@ -0,0 +1,266 @@
+// ==UserScript==
+// @name asus router interface Accessibility Fixes
+// @grant  unsafeWindow
+// @namespace      http://axSgrease.derekriemer.org/
+// @description    Improves the accessibility of the asus router management interface
+// @author         James Teh <jteh@mozilla.com>, derek riemer <git@derekriemer.com>
+// @copyright 2019-2024 Mozilla Corporation, Derek Riemer
+// @license Mozilla Public License version 2.0
+// @version        2024.1
+// @include http://asusrouter.com/*
+// @include http://www.asusrouter.com/*
+// ==/UserScript==
+
+/*** Functions for common tweaks. ***/
+
+/**
+ * Adds text to the given live region, and clears it a second later so it's no
+ * longer perceivable.
+ * @param {string} regionid an id of a region.
+ */
+function announce(text, regionId) {
+	getLiveRegion(regionId)
+		.then((region) => {
+			region.innerText = text;
+			setTimeout(() => {
+				region.innerText = '';
+			}, 1000);
+		});
+}
+
+/**
+ * create or fetch a live region that can be used with updateLiveRegion. Returns a promise with the region.
+ * @param {string} id the name of the new live region. This is an html id.
+ * @return {!Promise<HTMLElement>} a div that contains the live region. This can typically be ignored, this exxists to aid in chaining creation of non-existant regions.
```

Comment:
```suggestion
 * @return {!Promise<HTMLElement>} a div that contains the live region. This can typically be ignored, this exists to aid in chaining creation of non-existant regions.
```